sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java

```diff
@@ -119,24 +124,60 @@ private void analyze(
 
       SonarTSRunnerCommand command = executableBundle.getSonarTsRunnerCommand(tsconfigPath, inputFilesForThisConfig, typeScriptRules);
-      SonarTSRunnerResponse[] responses = executeExternalRunner(command, localTypescript);
-
-      for (SonarTSRunnerResponse response : responses) {
-        FileSystem fileSystem = sensorContext.fileSystem();
-        InputFile inputFile = fileSystem.inputFile(fileSystem.predicates().hasAbsolutePath(response.filepath));
-        if (inputFile != null) {
-          saveHighlights(sensorContext, response.highlights, inputFile);
-          saveSymbols(sensorContext, response.symbols, inputFile);
-          saveMetrics(sensorContext, response, inputFile);
-          saveCpd(sensorContext, response.cpdTokens, inputFile);
-          saveFailures(sensorContext, response.issues, typeScriptRules);
-        } else {
-          LOG.error("Failed to find input file for path `" + response.filepath + "`");
+      if (checkNodeVersion(executableBundle.getNodeExecutable())) {
+        SonarTSRunnerResponse[] responses = executeExternalRunner(command, localTypescript);
+
+        for (SonarTSRunnerResponse response : responses) {
+          FileSystem fileSystem = sensorContext.fileSystem();
+          InputFile inputFile = fileSystem.inputFile(fileSystem.predicates().hasAbsolutePath(response.filepath));
+          if (inputFile != null) {
+            saveHighlights(sensorContext, response.highlights, inputFile);
+            saveSymbols(sensorContext, response.symbols, inputFile);
+            saveMetrics(sensorContext, response, inputFile);
+            saveCpd(sensorContext, response.cpdTokens, inputFile);
+            saveFailures(sensorContext, response.issues, typeScriptRules);
+          } else {
+            LOG.error("Failed to find input file for path `" + response.filepath + "`");
+          }
         }
       }
     }
   }
 
+  private boolean checkNodeVersion(String nodeExecutable) {
+    LOG.debug("Checking node version");
+    String version = getNodeVersion(nodeExecutable);
+    if (version == null) {
+      return false;
+    }
+
+    Pattern versionPattern = Pattern.compile("v?(\\d+)\\.\\d+\\.\\d+");
+    Matcher versionMatcher = versionPattern.matcher(version);
+    if (versionMatcher.matches()) {
+      int major = Integer.parseInt(versionMatcher.group(1));
+      if (major < MIN_NODE_VERSION) {
+        LOG.error(String.format("Only Node.js v%s or later is supported, got %s", MIN_NODE_VERSION, version));
+        return false;
+      }
+    } else {
+      LOG.error(String.format("Failed to parse Node.js version, got %s", version));
+      return false;
+    }
+
+    LOG.debug(String.format("Using Node.js %s", version));
+    return true;
+  }
+
+  private @Nullable String getNodeVersion(String nodeExecutable) {
+    try {
+      Process process = Runtime.getRuntime().exec(nodeExecutable + " -v");
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Make sure "nodeExecutable" is properly sanitized before use in this OS command. ([squid:S2076](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2076))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=stas%2Fnode-version&issues=AWCicK8kqDrCo5ysgJkl&open=AWCicK8kqDrCo5ysgJkl)

[ISSUE_KEY]: AWCicK8kqDrCo5ysgJkl

Reply:
This issue has been resolved by a user in SonarQube

[ISSUE_STATUS]: RESOLVED